src/components/Modal/AuthorizeApplicationModalNewStaker/AuthorizationFormNewStaker.tsx

```diff
@@ -0,0 +1,129 @@
+import { Form } from "../../Forms"
+import { FC } from "react"
+import { FormikErrors, FormikProps, useField, withFormik } from "formik"
+import { getErrorsObj, validateAmountInRange } from "../../../utils/forms"
+import {
+  Alert,
+  AlertIcon,
+  Badge,
+  BodyMd,
+  BodyXs,
+  Box,
+  Button,
+  Card,
+  HStack,
+  LabelMd,
+} from "@threshold-network/components"
+import { useModal } from "../../../hooks/useModal"
+import AuthorizationCardNewStaker from "./AuthorizationCardNewStaker"
+
+export type FormValues = {
+  tbtcAmountToAuthorize: string | number
+  isTbtcChecked: boolean
+  randomBeaconAmountToAuthorize: string | number
+  isRandomBeaconChecked: boolean
+}
+
+export interface AuthInputConstraints {
+  min: string | number
+  max: string | number
+}
+
+interface Props {
+  onSubmitForm: (values: FormValues) => void
+  tbtcInputConstraints: AuthInputConstraints
+  randomBeaconInputConstraints: AuthInputConstraints
+}
+
+export const formikWrapper = withFormik<Props, FormValues>({
+  handleSubmit: (values, { props }) => {
+    props.onSubmitForm(values)
+  },
+  mapPropsToValues: (props) => ({
+    tbtcAmountToAuthorize: props.tbtcInputConstraints.max,
+    isTbtcChecked: true,
+    randomBeaconAmountToAuthorize: props.randomBeaconInputConstraints.max,
+    isRandomBeaconChecked: true,
+  }),
+  validate: (values, props) => {
+    const errors: FormikErrors<FormValues> = {}
+
+    errors.tbtcAmountToAuthorize = validateAmountInRange(
+      values?.tbtcAmountToAuthorize?.toString(),
+      props.tbtcInputConstraints.max.toString(),
+      props.tbtcInputConstraints.min.toString()
+    )
+
+    errors.randomBeaconAmountToAuthorize = validateAmountInRange(
+      values?.tbtcAmountToAuthorize?.toString(),
+      props.randomBeaconInputConstraints.max.toString(),
+      props.randomBeaconInputConstraints.min.toString()
+    )
```

Comment:
Validation for random beacon does not work. Should be:

```suggestion
    errors.randomBeaconAmountToAuthorize = validateAmountInRange(
      values?.randomBeaconAmountToAuthorize?.toString(),
      props.randomBeaconInputConstraints.max.toString(),
      props.randomBeaconInputConstraints.min.toString()
    )
```

Reply:
Fixed in ad1d9a5259b7a38cc9f1d99e5b50c671a0d6abb2